S.DS.P.psm1

```diff
@@ -714,6 +718,16 @@ More about System.DirectoryServices.Protocols: http://msdn.microsoft.com/en-us/l
         }
         $LdapConnection.SessionOptions.ProtocolVersion=$ProtocolVersion
 
+        if ($AcceptSelfSignedCert) {
+            $LdapConnection.SessionOptions.VerifyServerCertificate = { param(
+                    [Parameter(Mandatory)][DirectoryServices.Protocols.LdapConnection]$LdapConnection,
+                    [Parameter(Mandatory)][Security.Cryptography.X509Certificates.X509Certificate2]$Certificate
+                )
+                $script:LdapCertificate = $Certificate

```

Comment:
Let's get ready for more customizable validation, I suggest something like:
```ps
$validator = { param(
                    [Parameter(Mandatory)][DirectoryServices.Protocols.LdapConnection]$LdapConnection,
                    [Parameter(Mandatory)][Security.Cryptography.X509Certificates.X509Certificate2]$Certificate
                )
                [System.Security.Cryptography.X509Certificates.X509Chain] $chain = new-object System.Security.Cryptography.X509Certificates.X509Chain
                $chain.ChainPolicy.VerificationFlags = $CertificateValidationFlags
                $result = $chain.Build($Certificate)
                $script:LdapCertificate = $Certificate
                return $result
            }
```
Then you can just pass 'AllowUnknownCertificateAuthority' as CertificateValidationFlags to get desired result, and we're more ready to expand certi validation logic in the future.
Can you test this approach with your LDAP server and update pull request accordingly? I will be happy to make this enhancement part of vNext

Thanks,
Jiri